src/mail.sh

```diff
@@ -63,10 +67,42 @@ function validate_encryption()
   fi
 }
 
+# This function checks that the necessary options to use
+# the git send-email have been set
+function mail_verify()
+{
+  local count=0
+  local -a missing_conf
+  local set_confs
+
+  set_confs=$(git config --get-regexp 'user|sendemail\.smtp')
+  for config in "${essential_config_options[@]}"; do
+    if ! printf '%s\n' "$set_confs" | grep -cF "$config" - &> /dev/null; then
+      missing_conf[$count]="$config"
+      count=$((count + 1))
+    fi
+  done
+
+  for config in "${recommended_config_options[@]}"; do
+    if ! printf '%s\n' "$set_confs" | grep -cF "$config" - &> /dev/null; then
+      missing_conf[$count]="$config"
+      count=$((count + 1))
+    fi
+  done
```

Comment:
Is this config essential or not? If not, you probably want to separate `missing_conf` for this specific case and just warn the user. Does it make sense, or did I miss something?

Reply:
Yes, you are correct.